### DISCUSSION TITLE
USB keyboard and KVM switch

### DISCUSSION THREAD

**Discussion by cm68** [thehighnibble/thehighnibble.github.io#84] [S-132 add-on for IMSAI & Cromemco replicas]
my S-132 works great with my keyboard directly connected to the S-132, but if i connect it via my KVM switch, it's dead, even though on the other KVM ports, it runs fine (i'm typing on it now).   is there any firmware on the USB boardlet that might be upgradable to handle the KVM case?  perhaps it gets enumerated every time I connect it.

is this familiar to anybody?

--curt

**Comment by dmcnaugh** [comment]
I suspect your KVM is acting as a USB hub and not simply passing through the keyboard.
You can verify this by connecting the KVM and keyboard to a PC and looking at the USB details under your preferred OS.

**The adapter does not support connections via a hub.**

It would be technically possible with some further development, but currently not on the priority list.
If it does ever get done, then I have made provision in the S-132 (and VT132) to be able to flash new firmware into the USB-HID microcontroller.

**Comment by cm68** [comment]
Thanks for the tip!

Is the code for the ch559 published?  I'd like to take a crack at it.  I
just ordered a USB analyzer.

--curt
